### PR TITLE
fix(Button): change disabled style

### PR DESCRIPTION
**Pull request by luciatoma**
# Description

Improve style for `disabled` button state.
[JIRA task](https://ticketswap.atlassian.net/browse/TTN-1791)

### How to test

- Checkout this branch
- run `yarn`
- Check `disabled` states for `Button` component

### CODE REVIEW

**Comment by rppld** [comment]
We can basically remove all the specific `disabled` style declarations for the different variants if we move forward with this change. The only one that needs to stay is

```jsx
${props =>
    props.disabled &&
    css`
      opacity: 0.35;
      text-shadow: none;
    `};
```

**Review by rppld — CHANGES_REQUESTED** [review]
We can basically remove all the specific `disabled` style declarations for the different variants if we move forward with this change. The only one that needs to stay is

```jsx
${props =>
    props.disabled &&
    css`
      opacity: 0.35;
      text-shadow: none;
    `};
```

**Comment by luciatoma** [comment]
> We can basically remove all the specific `disabled` style declarations for the different variants if we move forward with this change. The only one that needs to stay is
> 
> ```js-jsx
> ${props =>
>     props.disabled &&
>     css`
>       opacity: 0.35;
>       text-shadow: none;
>     `};
> ```

Updated the PR 👍

File: src/components/Button/index.js [hunk 1]
```diff
@@ -133,29 +133,9 @@ const StyledButton = styled.button`
...
-
-  ${props =>
-    props.variant === 'inverted' &&
-    props.disabled &&
-    css`
-      background-color: white;
-      color: ${color.earthLighter};
+      opacity: 0.35;
+      pointer-events: none;
+      text-shadow: none;
```
Comment (rppld) [inline src/components/Button/index.js]: I think the text-shadow can stay for disabled buttons.

File: src/components/Button/index.js [hunk 2]
```diff
@@ -133,29 +133,9 @@ const StyledButton = styled.button`
...
-      color: ${color.marsLighterAlpha};
-    `};
-
-  ${props =>
-    props.variant === 'inverted' &&
-    props.disabled &&
-    css`
-      background-color: white;
-      color: ${color.earthLighter};
+      opacity: 0.35;
```
Comment (rppld) [inline src/components/Button/index.js]: I’ll have a look at this with @jelleoverbeek at the office, we might increase this a bit to `0.5`.

**Review by rppld — APPROVED** [review]

**Pull request merged**